worker/worker/ffmegtools.go

```diff
@@ -14,12 +16,42 @@ func getDuration(file string) (float64, error) {
 	return gjson.Get(probe, "format.duration").Float(), nil
 }
 
-func getCodec(file string) (string, error) {
+func getVideoCodec(file string) (string, error) {
 	probe, err := probe(file)
 	if err != nil {
 		return "", err
 	}
-	return gjson.Get(probe, "streams.0.codec_name").String(), nil
+	codecNumber := gjson.Get(probe, "streams.#").Int()
+	videoIndex := -1
+	for i := 0; i < int(codecNumber); i++ {
+		if gjson.Get(probe, fmt.Sprintf("streams.%d.codec_type", i)).String() == "video" {
+			videoIndex = int(gjson.Get(probe, fmt.Sprintf("streams.%d.index", i)).Int())
+			break
```

Comment:
Instead we could just return here

```suggestion
			return gjson.Get(probe, fmt.Sprintf("streams.%d.codec_name", i)).String(), nil
```